src/Utilities/ToolLocationHelper.cs

```diff
@@ -1166,6 +1138,96 @@ internal static string[] GetApiContractReferences(IEnumerable<ApiContract> apiCo
             return contractWinMDs.ToArray();
         }
 
+        private static void GetValueUsingMatchingSDKManifest(TargetPlatformSDK matchingSdk, string platformKey, Action<PlatformManifest> action)
```

Comment:
It's rather unusual for this method to take a Action rather than just return the value.  Why not have it return the manifest and have the caller then process it?  We don't use this kind of pattern anywhere else in this file.